File: app/src/main/java/com/chrisprime/primestationonecontrol/fragments/BaseFragment.java
package com.chrisprime.primestationonecontrol.fragments;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

import com.chrisprime.primestationonecontrol.PrimeStationOneControlApplication;
import com.chrisprime.primestationonecontrol.activities.PrimeStationOneControlActivity;
import com.chrisprime.primestationonecontrol.utilities.PreferenceStore;
import com.squareup.otto.Bus;

import javax.inject.Inject;

/**
 * Created by cpaian on 7/26/15.
 */
public class BaseFragment extends Fragment {
    protected Bus mEventBus = PrimeStationOneControlApplication.eventBus;

    @Inject
    PreferenceStore mPreferenceStore;

    @Override
    public void onResume() {
        super.onResume();
        mEventBus.register(this);
    }

    /**
     * Runs the Runnable on the UI thread and discard the runnable if the activity has been nulled out.
     *
     * @param runnable a Runnable to run if the activity isn't null
     */
    protected void safeRunOnUiThread(@NonNull Runnable runnable) {
        final Activity activity = getActivity();
        if (activity != null) {
            activity.runOnUiThread(runnable);
        }
    }

    /**
     * Runs the Runnable on the I/O thread and discard the runnable if the activity has been nulled out.
     *
     * @param runnable a Runnable to run if the activity isn't null
     */
    protected void safeRunOnIoThread(@NonNull Runnable runnable) {
        final PrimeStationOneControlActivity activity = (PrimeStationOneControlActivity) getActivity();
        if (activity != null) {
            activity.runOnIoThread(runnable);
        }
    }

    @Override
    public void onPause() {
        super.onPause();
        mEventBus.unregister(this);
    }
}

File: app/src/main/java/com/chrisprime/primestationonecontrol/fragments/BaseFragment.kt
package com.chrisprime.primestationonecontrol.fragments

import android.support.v4.app.Fragment
import com.chrisprime.primestationonecontrol.PrimeStationOneControlApplication
import com.chrisprime.primestationonecontrol.activities.PrimeStationOneControlActivity
import com.chrisprime.primestationonecontrol.utilities.PreferenceStore
import javax.inject.Inject

/**
 * Created by cpaian on 7/26/15.
 */
open class BaseFragment : Fragment() {
    protected var mEventBus = PrimeStationOneControlApplication.eventBus

    @Inject
    lateinit var mPreferenceStore: PreferenceStore

    override fun onResume() {
        super.onResume()
        mEventBus.register(this)
    }

    /**
     * Runs the Runnable on the UI thread and discard the runnable if the activity has been nulled out.

     * @param runnable a Runnable to run if the activity isn't null
     */
    protected fun safeRunOnUiThread(runnable: Runnable) {
        val activity = activity
        activity?.runOnUiThread(runnable)
    }

    /**
     * Runs the Runnable on the I/O thread and discard the runnable if the activity has been nulled out.

     * @param runnable a Runnable to run if the activity isn't null
     */
    protected fun safeRunOnIoThread(runnable: Runnable) {
        val activity = activity as PrimeStationOneControlActivity
        activity.runOnIoThread(runnable)
    }

    override fun onPause() {
        super.onPause()
        mEventBus.unregister(this)
    }
}
